public/app/features/admin/Users/UsersTable.tsx

```diff
@@ -0,0 +1,177 @@
+import { css } from '@emotion/css';
+import React, { useMemo } from 'react';
+import { DefaultSortTypes } from 'react-table';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import {
+  InteractiveTable,
+  CellProps,
+  Tooltip,
+  Icon,
+  useStyles2,
+  Tag,
+  Pagination,
+  Column,
+  VerticalGroup,
+  HorizontalGroup,
+} from '@grafana/ui';
+import { TagBadge } from 'app/core/components/TagFilter/TagBadge';
+import { UserDTO } from 'app/types';
+
+import { Avatar } from './Avatar';
+import { OrgUnits } from './OrgUnits';
+
+type Cell<T extends keyof UserDTO = keyof UserDTO> = CellProps<UserDTO, UserDTO[T]>;
+
+interface UsersTableProps {
+  users: UserDTO[];
+  showPaging?: boolean;
+  totalPages: number;
+  onChangePage: (page: number) => void;
+  currentPage: number;
+}
+
+export const UsersTable = ({ users, showPaging, totalPages, onChangePage, currentPage }: UsersTableProps) => {
+  const showLicensedRole = useMemo(() => users.some((user) => user.licensedRole), [users]);
+  const columns: Array<Column<UserDTO>> = useMemo(
+    () => [
+      {
+        id: 'avatarUrl',
+        header: '',
+        cell: ({ cell: { value } }: Cell<'avatarUrl'>) => <Avatar src={value} alt={'User avatar'} />,
+      },
+      {
+        id: 'login',
+        header: 'Login',
+        cell: ({ cell: { value } }: Cell<'login'>) => value,
+        sortType: 'string',
+      },
+      {
+        id: 'email',
+        header: 'Email',
+        cell: ({ cell: { value } }: Cell<'email'>) => value,
+        sortType: 'string',
+      },
+      {
+        id: 'name',
+        header: 'Name',
+        cell: ({ cell: { value } }: Cell<'name'>) => value,
+        sortType: 'string',
+      },
+      {
+        id: 'orgs',
+        header: 'Belongs to',
+        cell: OrgUnitsCell,
+        sortType: (a, b) => (a.original.orgs?.length || 0) - (b.original.orgs?.length || 0),
+      },
+      ...(showLicensedRole
+        ? [
+            {
+              id: 'licensedRole',
+              header: 'Licensed role',
+              cell: LicensedRoleCell,
+              // Needs the assertion here, the types are not inferred correctly due to the  conditional assignment
+              sortType: 'string' as DefaultSortTypes,
```

Comment:
if you need type assertion, use an eslint ignore comment rather than adding it to betterer.

Reply:
but also,  you don't need to the type assertion! Just do `sortType: 'string' as const,`